### PR TITLE
Added support for ARM

### PR DESCRIPTION
**Pull request by vonProteus**
ARM64 Support #6 
related to https://github.com/C9Glax/tranga-website/pull/116
https://hub.docker.com/layers/vonproteus/glax-tranga-api/latest/images/sha256-a1cb9859b33d83c271015c70be4a7f214914b5c770e276a6cb1b77ad7a3734fe?context=explore

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,29 +1,42 @@
 ﻿# syntax=docker/dockerfile:1
+ARG DOTNET=7.0
 
-FROM mcr.microsoft.com/dotnet/sdk:7.0 as build-env
+FROM mcr.microsoft.com/dotnet/runtime:$DOTNET AS base
+WORKDIR /publish
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
+ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium
+RUN apt-get update \
+  && apt-get install -y libx11-6 libx11-xcb1 libatk1.0-0 libgtk-3-0 libcups2 libdrm2 libxkbcommon0 libxcomposite1 libxdamage1 libxrandr2 libgbm1 libpango-1.0-0 libcairo2 libasound2 libxshmfence1 libnss3 chromium \
```
Comment (vonProteus) [inline Dockerfile:9]: added `chromium` to base image

File: Dockerfile [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
 COPY Tranga.sln /src
-RUN dotnet restore /src/Tranga/Tranga.csproj
-RUN dotnet publish -c Release -o /publish
+COPY CLI/CLI.csproj /src/CLI/CLI.csproj
+COPY Logging/Logging.csproj /src/Logging/Logging.csproj 
+COPY Tranga/Tranga.csproj /src/Tranga/Tranga.csproj
+RUN dotnet restore /src/Tranga.sln
+
+COPY . /src/
+RUN dotnet publish -c Release -o /publish -maxcpucount:1
```
Comment (vonProteus) [inline Dockerfile:23]: `-maxcpucount:1` GitHub Action while building arm imagae has problems without it

File: .github/workflows/docker-base.yml
Comment (vonProteus) [inline .github/workflows/docker-base.yml:1]: removed base image

File: Dockerfile-base
Comment (vonProteus) [inline Dockerfile-base:1]: moved to main `Dockerfile`

File: Tranga/MangaConnectors/ChromiumDownloadClient.cs [hunk 3]
```diff
@@ -14,38 +14,10 @@ internal class ChromiumDownloadClient : DownloadClient
...
-
-            };
-            if (!browserFetcher.CanDownloadAsync(ChromiumVersion).Result)
-            {
-                Log($"Can't download browser version {ChromiumVersion}");
-                throw new Exception();
-            }
-            await browserFetcher.DownloadAsync(ChromiumVersion);
-        }
-
```
Comment (vonProteus) [inline Tranga/MangaConnectors/ChromiumDownloadClient.cs:43]: chrome is a part of base image now

File: docker-compose.local.yaml
Comment (vonProteus) [inline docker-compose.local.yaml:1]: now you can spin up all by `docker compose -f docker-compose.local.yaml up --build`

File: Dockerfile [hunk 4]
```diff
@@ -1,29 +1,42 @@
 ﻿# syntax=docker/dockerfile:1
+ARG DOTNET=7.0
 
-FROM mcr.microsoft.com/dotnet/sdk:7.0 as build-env
+FROM mcr.microsoft.com/dotnet/runtime:$DOTNET AS base
+WORKDIR /publish
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
+ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium
```
Comment (vonProteus) [inline Dockerfile:7]: now Puppeteer will use this version of chrome

File: Dockerfile [hunk 5]
```diff
@@ -1,29 +1,42 @@
...
 USER $UNAME
 
 WORKDIR /publish
-COPY --from=build-env /publish .
+COPY --chown=1000:1000 --from=build-env /publish .
 USER 0
-RUN chown 1000:1000 /publish
-ENTRYPOINT ["dotnet", "/publish/Tranga.dll", "-f", "-c", "-l", "/usr/share/tranga-api/logs"]
+ENTRYPOINT ["dotnet", "/publish/Tranga.dll"]
+CMD ["-f", "-c", "-l", "/usr/share/tranga-api/logs"]
```
Comment (vonProteus) [inline Dockerfile:42]: when you split it to `ENTRYPOINT` and `CMD` you can override only arguments when ruining via `docker run`

File: Dockerfile [hunk 6]
```diff
@@ -1,29 +1,42 @@
...
 
-FROM mcr.microsoft.com/dotnet/sdk:7.0 as build-env
+FROM mcr.microsoft.com/dotnet/runtime:$DOTNET AS base
+WORKDIR /publish
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
+ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium
+RUN apt-get update \
+  && apt-get install -y libx11-6 libx11-xcb1 libatk1.0-0 libgtk-3-0 libcups2 libdrm2 libxkbcommon0 libxcomposite1 libxdamage1 libxrandr2 libgbm1 libpango-1.0-0 libcairo2 libasound2 libxshmfence1 libnss3 chromium \
+  && apt-get autopurge -y \
+  && apt-get autoclean -y
```
Comment (vonProteus) [inline Dockerfile:11]: now base image has fewer layers and its smaller

File: Tranga/MangaConnectors/ChromiumDownloadClient.cs (hunk 3 above)
Comment (C9Glax) [reply Tranga/MangaConnectors/ChromiumDownloadClient.cs:43]: Could you rename method to "StartBrowser" for readabilitys sake?

File: .github/workflows/docker-base.yml
Comment (C9Glax) [reply .github/workflows/docker-base.yml:1]: Makes sense. Was to speed up build time on frequent changes/tests

File: docker-compose.local.yaml
Comment (C9Glax) [reply docker-compose.local.yaml:1]: why --build?
Images are pre-built no? Or am I missing something.

File: Tranga/MangaConnectors/ChromiumDownloadClient.cs (hunk 3 above)
Comment (vonProteus) [reply Tranga/MangaConnectors/ChromiumDownloadClient.cs]: renamed

File: docker-compose.local.yaml
Comment (vonProteus) [reply docker-compose.local.yaml:1]: this is new file `docker-compose.local.yaml` which will build new image from source and run it convenient for testing

Comment (C9Glax) [reply docker-compose.local.yaml:1]: Ah I see, so for local testing we'd run --build, got it

**Review by C9Glax — APPROVED** [review]
alright looks good

**Comment by C9Glax** [comment]
Gonna merge to cuttingedge first, see if changes work

**Comment by vonProteus** [comment]
arm image is build only in master branch

**Comment by C9Glax** [comment]
Yeah sorry, worded that wrong, gonna test _locally_ with cuttingedge-branch, then merge into master

**Comment by vonProteus** [comment]
anyway now enabling arm images in other branches is only changing `platforms: linux/amd64` to `platforms: linux/amd64,linux/arm64`

**Pull request merged**